_posts/2021-09-07-dd_00003_logic_gates.md

```diff
@@ -0,0 +1,347 @@
+
+And it goes even further. Look at the following circuit:
+
+todo: Picture where same input to NAND (= NOT)
+
+This just a NAND gate where both inputs are connected with each other. This means that both inputs will always have the same value. The truth table of this special NAND gate is just the table of the normal NAND, but without the rows where the inputs are different:
+
+| A | B | Z |
+|---|---|---|
+| 0 | 0 | 1 |
+| 1 | 1 | 0 |
+
+As both inputs are always the same, we can view this NAND gate also as a single input gate; let's just call its input "A". The truth table remains unchanged (just column B disappeared):
+
+| A | Z |
+|---|---|
+| 0 | 1 |
+| 1 | 0 |
+
+Wait, that looks exactly like the TT of the NOT gate!
+-->
+
+Here's another more complex circuit, try to come up (todo: use something else
+than "come up") with its truth table:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-xor1-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+This is its truth table:
+
+| A   | B   | Z   |
+| --- | --- | --- |
+| 0   | 0   | 0   |
+| 0   | 1   | 1   |
+| 1   | 0   | 1   |
+| 1   | 1   | 0   |
+
+Do you notice a pattern in this truth table? The output is only "on" when
+exactly one input is turned on. And in fact, this behavior turns out to be so
+important in digital circuits, that there is even a dedicated logic gate for it:
+the **XOR** gate (XOR stands for "exclusive OR"). The XOR gate has the following
+symbol:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-xor-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+And, with this we have covered most of the basic logic gates! For those of you
+who like a little challenge, we have prepared the following circuit:
+
+<figure class="img-center" style="width:90%">
+  <img src="/assets/images/dd-00003-excercise.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+Try to come up (todo: "come up" is not the correct term here) with its truth
+table! We have prepared it for you:
+
+| A   | B   | C   | Z   |
+| --- | --- | --- | --- |
+| 0   | 0   | 0   | ?   |
+| 0   | 0   | 1   | ?   |
+| 0   | 1   | 0   | ?   |
+| 0   | 1   | 1   | ?   |
+| 1   | 0   | 0   | ?   |
+| 1   | 0   | 1   | ?   |
+| 1   | 1   | 0   | ?   |
+| 1   | 1   | 1   | ?   |
+
+We will present you the solution in our next post, so don't miss that!
+
+Should this be in this post?:
+
+- Boolean equations
+- Online Logic Sim
+
+## Wrap-up
+
+To conclude this post, let's summarize what we've learned.
+
+- **Logic gates** perform **logical operations**
+- The behaviour of a logic gate can be described using a **truth table**
+- We have learned the following basic logic gates:
+  - AND
+  - OR
+  - NOT
+  - NAND
+  - NOR
+  - XOR
+
```

Comment:
Actually it would be nice to have a cheat sheet with the gate picture, truth table and name. Would also fit inside the getting started repo as it is basic knowledge which is also interesting for other parts

Reply:
Not a bad idea. I will see if I can find time the next days to create one.

Reply:
Nice!